integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/FilteringJsonTest.java

```diff
@@ -0,0 +1,96 @@
+import static pl.allegro.tech.hermes.test.helper.builder.SubscriptionBuilder.subscriptionWithRandomName;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topicWithRandomName;
+
+public class FilteringJsonTest {
+
+    @RegisterExtension
+    public static final HermesExtension hermes = new HermesExtension();
+
+    @RegisterExtension
+    public static final TestSubscribersExtension subscribers = new TestSubscribersExtension();
+
+    private static final MessageFilterSpecification MESSAGE_NAME_FILTER =
+            new MessageFilterSpecification(of("type", "jsonpath", "path", ".name", "matcher", "^Bob.*"));
+
+    private static final MessageFilterSpecification MESSAGE_COLOR_FILTER =
+            new MessageFilterSpecification(of("type", "jsonpath", "path", ".favoriteColor", "matcher", "grey"));
+
+    static final AvroUser BOB = new AvroUser("Bob", 50, "blue");
+    static final AvroUser ALICE = new AvroUser("Alice", 20, "magenta");
+    static final AvroUser ALICE_GREY = new AvroUser("Alice", 20, "grey");
+    static final AvroUser BOB_GREY = new AvroUser("Bob", 50, "grey");
+
+    private static final SubscriptionPolicy SUBSCRIPTION_POLICY = new SubscriptionPolicy(100, 2000, 1000, 1000, true, 100, null, 0, 1, 600);
+
+    @Test
+    public void shouldFilterIncomingEvents() {
+        // given
+        TestSubscriber subscriber = subscribers.createSubscriber();
+        Topic topic = hermes.initHelper().createTopic(topicWithRandomName().build());
+        hermes.initHelper().createSubscription(subscriptionWithRandomName(topic.getName(), subscriber.getEndpoint())
+                .withSubscriptionPolicy(SUBSCRIPTION_POLICY)
+                .withFilter(MESSAGE_NAME_FILTER)
+                .build());
+
+        // when
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), ALICE.asJson());
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), BOB.asJson());
+
+        // then
+        subscriber.waitUntilReceived(BOB.asJson());
+    }
+
+    @Test
+    public void shouldChainFilters() {
+        // given
+        TestSubscriber subscriber = subscribers.createSubscriber();
+        Topic topic = hermes.initHelper().createTopic(topicWithRandomName().build());
+        hermes.initHelper().createSubscription(subscriptionWithRandomName(topic.getName(), subscriber.getEndpoint())
+                .withSubscriptionPolicy(SUBSCRIPTION_POLICY)
+                .withFilter(MESSAGE_NAME_FILTER)
+                .withFilter(MESSAGE_COLOR_FILTER)
+                .build());
+
+        // when
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), ALICE.asJson());
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), ALICE_GREY.asJson());
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), BOB.asJson());
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), BOB_GREY.asJson());
+
+        // then
+        subscriber.waitUntilReceived(BOB_GREY.asJson());
```

Comment:
I'd say this check is insufficient because it only checks whether `BOB_GREY` has been received and does not fail when receiving other messages. Can we add an assertion that only this message was received? This also applies to other filtering tests